src/main/java/com/cburch/logisim/fpga/hdlgenerator/HDL.java

```diff
@@ -79,6 +135,69 @@ public static String xorOperator() {
     return isVHDL() ? " XOR " : "^";
   }
 
+  public static String addOperator(String signalOne, String signalTwo, boolean signed) {
+    return (isVHDL() ? "std_logic_vector(" : "")
+            + typecast(signalOne, signed)
+            + " + "
+            + typecast(signalTwo, signed)
+            + (isVHDL() ? ")" : "");
+  }
+  
+  public static String subOperator(String signalOne, String signalTwo, boolean signed) {
+    return (isVHDL() ? "std_logic_vector(" : "")
+            + typecast(signalOne, signed)
+            + " - "
+            + typecast(signalTwo, signed)
+            + (isVHDL() ? ")" : "");
+  }
+
+  public static String shiftlOperator(String signal, int with, int nrOfBits, boolean arithmetic) {
+    if (nrOfBits == 0)
+      return signal;
+    return isVHDL() ? LineBuffer.format("{{1}}{{2}} & {{4}}{{3}}{{4}}", signal, splitVector(with - 1 - nrOfBits, 0), "0".repeat(nrOfBits), nrOfBits == 1 ? "'" : "\"")
+                    : LineBuffer.format("{{{1}}{{2}},{{{3}}{1'b0}}}", signal, splitVector(with - 1 - nrOfBits, 0), nrOfBits);
```

Comment:
(`{{{3}}{1'b0}`) isn't working.
(`#E006: No mapping for '6{1'b0' placeholder in '{{6{1'b0}},test_srl[5:4] }'`)
@MarcinOrlowski, is there a way to make it work?

Reply:
You got triple opening `{` whichbcauses this.